src/PageLayout/useStickyPaneHeight.ts

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import {useInView} from 'react-intersection-observer'
+
+/**
+ * Calculates the height of the sticky pane such that it always
+ * fits into the viewport even when the header or footer are visible.
+ */
+// TODO: Handle sticky header
+export function useStickyPaneHeight() {
+  // Default the height to the viewport height
+  const [height, setHeight] = React.useState('100vh')
```

Comment:
@colebemis we should use `100vh` by default, but if the browser supports 100dvh, we should use that instead. `dvh` units take into consideration any dynamic browser viewport resizing. For example:

![image](https://user-images.githubusercontent.com/293280/183569309-9a2f34af-2970-4cdd-8b4c-d5a7bdf2922d.png)

([Source](https://twitter.com/jensimmons/status/1499441043930062854))

This is how I define it in the Primer CSS [implementation](https://github.com/primer/css/blob/main/src/layout/page-layout.scss#L114-L124):

```scss
.PageLayout-pane--sticky {
  position: sticky;
  top: 0;
  max-height: 100vh;
  overflow: auto;
  scrollbar-width: thin;

  @supports (max-height: 100dvh) {
    max-height: 100dvh;
  }
}
```

(Also note I'm using `max-height`, not `height`. Not sure how that'd compare to your implementation)

Reply:
Since dynamic viewport units are only supported in mobile browsers right now, and we disable sticky behavior at narrow viewports, I'm going to postpone using `dvh` for now

Reply:
@colebemis Safari on iPad also has that behavior.

Reply:
Ah good point, @vdepizzol. I'll see how we can use `dvh` units with this implementation.

Reply:
This PR is getting a little too big. I'll break it into a new PR. Just made a tracking issue: https://github.com/github/primer/issues/1211